docs/architecture/deep-dives/autofill/autofill-menu.md

```diff
@@ -0,0 +1,299 @@
+necessitating communication with the extension background through postMessage calls routed through
+the iframe element's parent. This ensures that any actions needing to occur within the extension
+background script cannot be initiated outside the
+[isolated context of the content script](https://developer.chrome.com/docs/extensions/mv3/content_scripts/#isolated_world)
+that injects the Auto-fill Menu UI elements into the DOM.
+
+These pages are rendered with a strict content security policy that prohibits the execution of any
+inline scripts or scripts foreign to the extension. This ensures that the pages cannot execute any
+form of script not explicitly defined within the extension, whether inline or through a script tag.
+
+### Passing Messages Between the Auto-fill Menu and the Extension
+
+As a consequence of the sandboxed iframe implementation, an indirect messaging system was necessary
+to enable communication between the extension background and the Auto-fill Menu pages. This
+messaging approach operates through the use of `postMessage` calls exchanged among the Auto-fill
+Menu pages, the content script, and the extension background.
+
+When an Auto-fill Menu page needs to communicate with the extension background, it initiates a
+`postMessage` call to its parent frame, the content script that injected the Auto-fill Menu into the
+webpage. This content script verifies the origin of the window message to confirm it originates from
+the expected source. Upon validation, it relays a message to the extension background via a
+long-lived port connection. The extension background, upon receiving this message, executes the
+necessary background logic and responds with a port message back to the content script. This message
+is then transferred to the Auto-fill Menu page through another `postMessage` call.
+
+Below is an illustration that demonstrates how this messaging system operates when initializing and
+retrieving cipher data for the Auto-fill Menu list page:
+
+[![Auto-fill Menu messaging](./assets/autofill-overlay-messaging.jpg) _(A visual representation of the messaging workflow of the Auto-fill Menu)_](./assets/autofill-overlay-messaging.jpg)
+
+As illustrated in the example above, any messages that need to be relayed from the `OverlayListPage`
+to the `OverlayBackground` script must pass through the `AutofillOverlayIframe` content script. This
+arrangement ensures that the Auto-fill Menu pages cannot directly access the extension background
+script and, likewise, that the extension background script is not directly manipulable by the
+Auto-fill Menu pages.
+
+### Populating the Auto-fill Menu with Data
+
+We exercised considerable selectivity in the types of data passed to the Auto-fill Menu to safeguard
+against compromising a user's vault. Consequently, only the following data types are transmitted to
+the Auto-fill Menu UI elements:
+
+```typescript
```

Comment:
⚠️ I don't see syntax highlighting rendering here, or my eyes are off. Perhaps not supported with our engine selections.

Reply:
Hmm, it appears to be working on my side within Dark mode, even if it isn't working in light mode.


<img width="824" alt="Screenshot 2024-01-02 at 1 14 48 PM" src="https://github.com/bitwarden/contributing-docs/assets/16629865/77073715-6f5f-4c7c-828e-fd93b1e6eaf1">

This is dependent on Docusaurs to some degree.... but not a huge issue from my perspective given that it's a simple type definition.